neural_compressor/compression/pruner/pruners/__init__.py

```diff
@@ -0,0 +1,90 @@
+"""pruning patterns."""
+# !/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2022 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from os.path import dirname, basename, isfile, join
+import glob
+from .base import PRUNERS
+from ..criteria import CRITERIA
+
+modules = glob.glob(join(dirname(__file__), "*.py"))
+
+for f in modules:
+    if isfile(f) and not f.startswith('__') and not f.endswith('__init__.py'):
+        __import__(basename(f)[:-3], globals(), locals(), level=1)
+
+FRAMEWORK = {
+    'pytorch': 'pt',
+    'keras': 'keras'
+}
+
+
+def parse_valid_pruner_types():
+    """Get all valid pruner names."""
+    valid_pruner_types = []
+    for x in CRITERIA.keys():
```

Comment:
There may be new pruner types added later that don't have a corresponding criterion, and this restriction may be a bit strict.

Reply:
This func is used to show the valid pruner name. When we add a new criteria, we use register_criterion as a decorator and store the class name into a hyper parameter CRITERIA. Therefore, new criteria can be enter though CRITERIA.keys().